File: app/src/main/java/com/extremewakeup/soundalarm/model/Alarm.kt
package com.extremewakeup.soundalarm.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.time.LocalTime

@Entity
@Serializable
data class Alarm(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    @Serializable(with = LocalTimeSerializer::class) val time: LocalTime,
    val daysActive: List<String>,
    val volume: Int,
    var isActive: Boolean,
    val userId: Int
)

File: scripts/ExtremeAlarm/src/Alarm.cpp
#include <vector>
#include <ArduinoJson.h>

struct Alarm {
    int id;
    String time;  // Format HH:MM for simplicity
    std::vector<String> daysActive;  // Using vector for simplicity, ensure C++ STL is enabled
    int volume;
    bool isActive;
    int userId;
};

// Define a global list (vector) of alarms
std::vector<Alarm> alarms;
